specs/_features/eip7549/beacon-chain.md

```diff
@@ -0,0 +1,138 @@
+- [Helper functions](#helper-functions)
+  - [Misc](#misc)
+    - [`get_committee_indices`](#get_committee_indices)
+  - [Beacon state accessors](#beacon-state-accessors)
+    - [Modified `get_attesting_indices`](#modified-get_attesting_indices)
+  - [Block processing](#block-processing)
+    - [Modified `process_attestation`](#modified-process_attestation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification to move the attestation committee index outside of the signed message. For motivation, refer to [EIP-7549](https://eips.ethereum.org/EIPS/eip-7549).
+
+*Note:* This specification is built upon [Deneb](../../deneb/beacon_chain.md) and is under active development.
+
+## Preset
+
+| Name | Value | Description |
+| - | - | - |
+| `MAX_ATTESTER_SLASHINGS` | `2**0` (= 1) |
+| `MAX_ATTESTATIONS`       | `2**3` (= 8) |
+
+## Containers
+
+### Modified containers
+
+#### `Attestation`
+
+```python
+class Attestation(Container):
+    aggregation_bits: List[Bitlist[MAX_VALIDATORS_PER_COMMITTEE], MAX_COMMITTEES_PER_SLOT]  # [Modified in EIP7549]
+    data: AttestationData
+    committee_bits: Bitvector[MAX_COMMITTEES_PER_SLOT]  # [New in EIP7549]
+    signature: BLSSignature
+```
+
+#### `IndexedAttestation`
+
+```python
+class IndexedAttestation(Container):
+    # [Modified in EIP7549]
+    attesting_indices: List[ValidatorIndex, MAX_VALIDATORS_PER_COMMITTEE * MAX_COMMITTEES_PER_SLOT]
+    data: AttestationData
+    signature: BLSSignature
+```
+
+## Helper functions
+
+### Misc
+
+#### `get_committee_indices`
+
+```python
+def get_committee_indices(commitee_bits: Bitvector) -> List[CommitteeIndex]:
+    return [CommitteeIndex(index) for bit, index in enumerate(commitee_bits) if bit]
+```
+
+### Beacon state accessors
+
+#### Modified `get_attesting_indices`
+
+```python
+def get_attesting_indices(state: BeaconState, attestation: Attestation) -> Set[ValidatorIndex]:
+    """
+    Return the set of attesting indices corresponding to ``aggregation_bits`` and ``committee_bits``.
+    """
+
+    output = set()
+    committee_indices = get_committee_indices(attestation.committee_bits)
+    for index in committee_indices:
+        attesting_bits = attestation.aggregation_bits[index]
+        committee = get_beacon_committee(state, attestation.data.slot, index)
+        committee_attesters = set(index for i, index in enumerate(committee) if attesting_bits[i])
+        output = output.union(committee_attesters)
+
+    return output
+```
+
+### Block processing
+
+#### Modified `process_attestation`
+
+```python
+def process_attestation(state: BeaconState, attestation: Attestation) -> None:
+    data = attestation.data
+    assert data.target.epoch in (get_previous_epoch(state), get_current_epoch(state))
+    assert data.target.epoch == compute_epoch_at_slot(data.slot)
+    assert data.slot + MIN_ATTESTATION_INCLUSION_DELAY <= state.slot
```

Comment:
Why is the max age removed here? (`state.slot <= data.slot + SLOTS_PER_EPOCH`)

https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#attestations

```python
assert data.slot + MIN_ATTESTATION_INCLUSION_DELAY <= state.slot <= data.slot + SLOTS_PER_EPOCH
```

Reply:
It is initially removed in https://github.com/ethereum/consensus-specs/pull/3360

Reply:
Oh, thanks for the pointer! Have missed the Deneb specs while reviewing.